Assets/HoloToolkit/Input/Scripts/GameControllerManipulator.cs

```diff
@@ -0,0 +1,160 @@
+            {
+                rz = RotateAroundAxis(go, RotateAroundZAxisName, cameraTransform.forward, RotateAroundZAxisSpeed);
+                ry = RotateAroundAxis(go, RotateAroundYAxisName, cameraTransform.up, RotateAroundYAxisSpeed);
+                rx = RotateAroundAxis(go, RotateAroundXAxisName, cameraTransform.right, RotateAroundXAxisSpeed);
+            }
+
+            //Move
+            bool usingSameAxisForRotateAndMove = RotateAroundYAxisName == MoveXAxisName ||
+                                                 RotateAroundYAxisName == MoveZAxisName || 
+                                                 RotateAroundZAxisName == MoveXAxisName ||
+                                                 RotateAroundZAxisName == MoveYAxisName;
+
+            float x = 0;
+            float y = 0;
+
+            //only move if the movement axis has not been used for rotation
+            if (!(enableRotate && usingSameAxisForRotateAndMove))
+            {
+                x = Input.GetAxis(MoveXAxisName) * MoveXAxisSpeed * 60 * Time.deltaTime;
+                go.transform.RotateAround(cameraTransform.position, cameraTransform.up, x);
+
+                y = Input.GetAxis(MoveYAxisName) * MoveYAxisSpeed * 60 * Time.deltaTime;
+                go.transform.RotateAround(cameraTransform.position, cameraTransform.right, y);
+            }
+
+            var z = Input.GetAxis(MoveZAxisName) * MoveZAxisSpeed * 60 * Time.deltaTime;
+            go.transform.position += cameraTransform.forward*z*-0.03f;
+
+            var stickInFrontOfMe = Input.GetButton(MoveWithGazeButtonName);
+
+            if (stickInFrontOfMe)
+            {
+                if (stickDistance == 0)
+                {
+                    stickDistance = (go.transform.position - cameraTransform.position).magnitude;
+                }
+
+                go.transform.position = cameraTransform.position + cameraTransform.forward * stickDistance;
+            }
+            else
+            {
+                stickDistance = 0;
+            }
+
+
+            if (stickInFrontOfMe || x != 0 || y != 0 || z != 0 || rz != 0 || ry != 0 || rx != 0)
+            {
+                lastAffectedObject = go;
+            }
+#if !UNITY_EDITOR
+            else
+            {
+                lastAffectedObject = null; //easier to test in Unity this way
+            }
+#endif
+        }
+
+        /// <summary>
+        /// Rotates the specified GameObject as dictated by joystick values and multipliers
+        /// </summary>
+        /// <param name="go">the gameObject to rotate</param>
+        /// <param name="joyAxisName">the name of the joystick axis</param>
+        /// <param name="vectorToRotateAround">vector to rotate around</param>
+        /// <param name="speed">rotation speed</param>
+        /// <returns>the amount of rotation applied</returns>
+        private float RotateAroundAxis(GameObject go, string joyAxisName, Vector3 vectorToRotateAround, float speed)
```

Comment:
Nit: Instead of just go, could we rename this to be a bit descriptive? Like `rotateGo` or `goToRotate`?